Mage/src/main/java/mage/players/PlayerImpl.java

```diff
@@ -4674,6 +4679,29 @@ public boolean isDrawsOnOpponentsTurn() {
         return drawsOnOpponentsTurn;
     }
 
+    @Override
+    public int getSpeed() {
+        return speed;
+    }
+
+    @Override
+    public void initSpeed(Game game) {
+        if (speed > 0) {
+            return;
+        }
+        speed = 1;
+        game.getState().addDesignation(new Speed(), game, getId());
+        game.informPlayers(this.getLogName() + "'s speed is now 1.");
+    }
+
+    @Override
+    public void increaseSpeed(Game game) {
+        if (speed < 4) {
+            speed++;
+            game.informPlayers(this.getLogName() + "'s speed has increased to " + speed);
```

Comment:
`increaseSpeed` called by layer effects too (`StartYourEnginesEffect`) and can send to client side unfinished game state (in the middle of the game state calculation, e.g. before ApplyEffects finished). I recommend to remove that inform message from layer effects or call with param like `increaseSpeed(Game game, boolean informPlayers)`

Reply:
I'm actually gonna change it to being a state-based action so it shouldn't matter